frontend/src/components/SimpleRange/SimpleRange.js

```diff
@@ -2,38 +2,46 @@ import { styled } from '@mui/material/styles'
 import LinearProgress, { linearProgressClasses } from '@mui/material/LinearProgress'
 import PropTypes from 'prop-types'
 
-const GradientProgress = styled(LinearProgress)(({ theme, value }) => ({
-  [`& .${linearProgressClasses.bar}`]: {
-    overflow: 'hidden',
-    backgroundColor: 'transparent',
-    borderRadius: 1,
-    '&:before': {
-      content: '""',
-      display: 'block',
-      width: '100%',
-      height: '100%',
-      borderRadius: theme.shape.borderRadius,
-      transition: 'transform .4s linear',
-      transform: `translateX(${100 - value}%)`,
-      background: `linear-gradient(270deg, ${theme.palette.canary.main} 0%, ${theme.palette.chestnutRose.main} 50%, ${theme.palette.ultramarine.main} 100%)`, // eslint-disable-line
+const nonDomPropsList = ['colorScheme']
+const GradientProgress = styled(LinearProgress, { shouldForwardProp: name => !nonDomPropsList.includes(name) })(
+  ({ theme, value, colorScheme }) => ({
+    [`& .${linearProgressClasses.bar}`]: {
+      overflow: 'hidden',
+      backgroundColor: 'transparent',
+      borderRadius: 1,
+      '&:before': {
+        content: '""',
+        display: 'block',
+        width: '100%',
+        height: '100%',
+        borderRadius: theme.shape.borderRadius,
+        transition: 'transform .4s linear',
+        transform: `translateX(${100 - value}%)`,
+        background: `linear-gradient(270deg, ${colorScheme?.[0] || theme.palette.canary.main} 0%, ${
+          colorScheme?.[1] || theme.palette.chestnutRose.main
+        } 50%, ${colorScheme?.[2] || theme.palette.ultramarine.main} 100%)`, // eslint-disable-line
```

Comment:
1. Instead of reversing the order of colorScheme in Map.js, why not apply them in reverse order here? Or rotate the gradient 180° and swap the order of the default colors?

2. There could be more or fewer than 3 colors, and their stops might not be evenly spaced. Like in `buildColorExpression`, the metric's `domain` and its `colorScheme` need to be interwoven to create the correct gradient. This domain's colors happen to be evenly distributed:
    ```json
    {
      "metric": {
        "domain": [0, 0.6, 1.2]
      },
      "paint": {
        "default": {
          "fill": {
            "colorScheme": ["#fff5eb", "#fd8d3c", "#7f2704"],
          },
        }
      }
    }
    ```
    And become:
    ```css
    background: linear-gradient(90deg, #fff5eb 0%, #fd8d3c 50%, #7f2704 100%);
    ```
    But this domain's colors are squeezed at the ends:
    ```json
    {
      "metric": {
        "domain": [0, 2, 8, 10]
      },
      "paint": {
        "default": {
          "fill": {
            "colorScheme": ["#00ffff", "#ffff00", "#ff0000", "#ff00ff"],
          },
        }
      }
    }
    ```
    And become:
    ```css
    background: linear-gradient(90deg, #00ffff 0%, #ffff00 20%, #ff0000 80%, #ff00ff 100%);
    ```